x/poolmanager/README.md

```diff
@@ -478,8 +478,34 @@ type TakerFeeParams struct {
 
 Not shown here is a separate KVStore, which holds overrides for the defaultTakerFee.
 
-At time of swap, all taker fees are sent to the `taker_fee_collector` module account. At epoch:
+The Osmosis protocol now supports setting up taker fee agreements with specific denoms to share a certain percentage of taker fees generated in any route containing those denoms:
 
+```go
+func (k *Keeper) SetTakerFeeShareAgreementForDenom(ctx sdk.Context, takerFeeShare types.TakerFeeShareAgreement) error
+
+type TakerFeeShareAgreement struct {
+    Denom       string `json:"denom"`
+    SkimAddress string `json:"skim_address"`
+    SkimPercent string `json:"skim_percent"`
+}
+```
+
+For example, if the agreement specifies a 10% `skim_percent`, then 10% of all taker fees generated in a swap route containing the specified denom will be sent to the `skim_address` at the end of each epoch. These percentages are additive, so if there are agreements with skim percents of 10%, 20%, and 30%, the total skim percent for the route will be 60%. If the skim percent exceeds 100%, the swap fails to go through.
+
+The protocol can also register alloyed asset pools, which are pools containing denoms with taker fee share agreements:
+
+```go
+func (k *Keeper) SetRegisteredAlloyedPool(ctx sdk.Context, poolId uint64) error
+```
+
+If a swap route contains an alloyed asset pool but no individual taker fee share agreement denoms, the taker fees will be skimmed based on the underlying denoms in the pool that have taker fee share agreements, adjusted by their respective weights. For example, if an alloyed asset pool contains 50% nBTC (10% skim percent) and 50% iBTC (5% skim percent), the total skim percent for the pool will be 7.5%, with 5% of the taker fees going to the nBTC skim address and 2.5% to the iBTC skim address. Just like the taker fee share agreements, if the skim percent exceeds 100%, the swap fails to go through.
+
+NOTE: We cache the alloyed pool asset weight composition and recalculate in the poolmanager EndBlock every 700 blocks (approx 30 minutes at 2.6s blocks) to avoid recalculating the weights for every swap.
```

Comment:
Is this a gov controlled parameter? Might be helpful to have it so if not for more flexibility

Reply:
Commented on this here, lmk what you think: https://github.com/osmosis-labs/osmosis/pull/8310#discussion_r1690828920